src/Core/Framework/Webhook/Hookable/HookableEventCollector.php

```diff
@@ -99,6 +84,30 @@ public function getEntityWrittenEventNamesWithPrivileges(): array
         return $entityWrittenEventNames;
     }
 
+    /**
+     * Dynamically discovers all hookable entities by checking for services tagged with 'shopware.entity.hookable'.
+     *
+     * @return array<string>
+     */
+    public function getHookableEntities(): array
+    {
+        if ($this->hookableEntities !== null) {
+            return $this->hookableEntities;
+        }
+
+        $hookableEntities = [];
+
+        foreach ($this->hookableEntityDefinitions as $definition) {
+            if ($definition instanceof EntityDefinition) {
+                $hookableEntities[] = $definition->getEntityName();
+            }
+        }
+
```

Comment:
I have not tested it yet, but you might want to consider giving it a try. The idea is to tag the Entity that has an Entity Attribute.

```suggestion
...
use Shopware\Core\Framework\DataAbstractionLayer\Entity;
use Shopware\Core\Framework\DataAbstractionLayer\Attribute\Entity as EntityAttribute;
...


if ($definition instanceof Entity) {

	    $reflection = new \ReflectionClass(get_class($definition));
	
	    $collection = $reflection->getAttributes(EntityAttribute::class);
	
	    if (empty($collection)) {
	        return [];
	    }
	
	    /** @var EntityAttribute $instance */
	    $instance = $collection[0]->newInstance();
	    $hookableEntities[] = $instance->name;
	}
```